src/flexcomm.rs

```diff
@@ -103,7 +163,23 @@ macro_rules! impl_flexcomm {
 				   unsafe { w.mult().bits(0) });
 
 			enable_and_reset::<[<FLEXCOMM $idx>]>();
+
+            FlexcommRef::new::<Self>()
 		    }
+
+            fn disable() {
+            // SAFETY: safe from single executor
+			let clkctl1 = unsafe { crate::pac::Clkctl1::steal() };
+            clkctl1.flexcomm($idx).fcfclksel().write(|w| w.sel().none());
+            clkctl1.flexcomm($idx).frgclksel().write(|w| w.sel().none());
+            disable::<[<FLEXCOMM $idx>]>();
+            }
+
+            #[allow(private_interfaces)]
+            fn state() -> &'static State {
+            static STATE: State = State::new();
+            &STATE
+            }
```

Comment:
please run `cargo +nightly fmt` on your code.

Reply:
For me `cargo +nightly fmt` does not touch this code. Because this segment is in a `paste!` block, `fmt` seems to struggle with it. Honestly the original code also was formatted incorrectly, and I mimicked it thinking there was maybe a good reason for it.

If for you `fmt` works fine, please let me know. Meanwhile I'll fix the formatting by hand.